clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -324,11 +324,14 @@ public void disconnect(String nodeId) {
         log.info("Client requested disconnect from node {}", nodeId);
         selector.close(nodeId);
         long now = time.milliseconds();
-        cancelInFlightRequests(nodeId, now, abortedSends);
+        cancelInFlightRequests(nodeId, now, abortedSends, false);
         connectionStates.disconnected(nodeId, now);
     }
 
-    private void cancelInFlightRequests(String nodeId, long now, Collection<ClientResponse> responses) {
+    private void cancelInFlightRequests(String nodeId,
```

Comment:
nit: final-lize all params

Reply:
I didn't want to introduce any inconsistency in this change as the rest of the code doesn't follow that pattern.

Reply:
I'm marking this as "resolved" for now. Please reopen in protest if you strongly disagree 😄